app/src/main/java/org/mozilla/focus/telemetry/TelemetryWrapper.kt

```diff
@@ -306,6 +317,7 @@ object TelemetryWrapper {
         TelemetryHolder.get()
                 .queuePing(TelemetryCorePingBuilder.TYPE)
                 .queuePing(TelemetryEventPingBuilder.TYPE)
+                .also { if (shouldSendMobileMetrics) it.queuePing(TelemetryMobileMetricsPingBuilder.TYPE) }
                 .scheduleUpload()
```

Comment:
I would expect the decision about whether to send a ping or not to be made here (including all the timestamp code) instead of somewhere outside and then passed into this class.

It would be really nice if we could collect the data here too. But I assume this is hard because of the asynchronous API? Alternatively we could collect the data from startSession() and queue the ping right there.